File: src/com/falconraptor/utilities/colors.java
package com.falconraptor.utilities;

import com.falconraptor.utilities.logger.Logger;

public class Colors {
    private static String log = "[com.falconraptor.utilities.Colors.";

    public static int[] checkerrors(String test) {
        return new int[]{findnum(test, "r="), findnum(test, "g="), findnum(test, "b=")};
    }

    public static int findnum(String abc, String find) {
        int a = 5, t;
        for (int i = 0; i < 4; i++) {
            try {
                t = Integer.parseInt(abc.substring(abc.indexOf(find) + 2, abc.indexOf(find) + a));
                break;
            } catch (Exception e) {
                if (Logger.level <= 5) Logger.logERROR(log + "findnum] " + e);
                a--;
            }
        }
        return a;
    }
}

File: src/com/falconraptor/utilities/Colors.scala
package com.falconraptor.utilities

import com.falconraptor.utilities.logger.Logger

object Colors {
  def checkErrors(test: String): Array[Int] = {
    Array[Int](findNum(test, "r="), findNum(test, "g="), findNum(test, "b="))
  }

  def findNum(abc: String, find: String): Int = {
    var a = 5
    var i = 0
    while (i < 4) {
      try {
        abc.substring(abc.indexOf(find) + 2, abc.indexOf(find) + a).toInt
        i = 4
      } catch {
        case e: Exception =>
          val log: String = "[com.falconraptor.utilities.Colors."
          if (Logger.level <= 5) Logger.logERROR(log + "findNum] " + e)
          a -= 1
      }
      i += 1
    }
    a
  }
}